code/game/objects/items/devices/pcmc.dm

```diff
@@ -0,0 +1,258 @@
+	var/autorefreshing = FALSE
+	var/transmitting = FALSE
+	var/injuryonly = FALSE
+	var/fullmode = TRUE
+
+/obj/item/device/pcmc/New()
+	..()
+	update_icon()
+
+/obj/item/device/pcmc/Destroy()
+	..()
+
+/obj/item/device/pcmc/update_icon()
+	overlays.Cut()
+	if(emped)
+		overlays += image(icon, "pcmcemp")
+		return
+	if(transmitting)
+		overlays += image(icon, "pcmcon")
+
+/obj/item/device/pcmc/emp_act(severity)
+	emped = TRUE
+	transmitting = FALSE
+	set_light(0)
+	update_icon()
+	SStgui.update_uis(src)
+	spawn(120 SECONDS)
+		emped = FALSE
+		update_icon()
+		SStgui.update_uis(src)
+
+/obj/item/device/pcmc/attack_self(mob/user)
+	tgui_interact(user)
+
+/obj/item/device/pcmc/examine(mob/user)
+	if(Adjacent(user) || isobserver(user))
+		attack_self(user)
+	else
+		..()
+
+/obj/item/device/pcmc/proc/get_location_name()
+	var/turf/device_turf = get_turf(src)
+	var/area/device_area = get_area(src)
+	if (emped)
+		return "ERROR"
+	else if(!device_turf || !device_area)
+		return "UNKNOWN"
+	else if(device_turf.z > WORLD_X_OFFSET.len)
+		return "[format_text(device_area.name)] (UNKNOWN, UNKNOWN, UNKNOWN)"
+	else
+		return "[format_text(device_area.name)] ([device_turf.x-WORLD_X_OFFSET[device_turf.z]], [device_turf.y-WORLD_Y_OFFSET[device_turf.z]], [device_turf.z])"
+
+
+/obj/item/device/pcmc/proc/get_crew()
+	//looping though carbons
+	var/list/crewlist = list()
+	for(var/mob/living/carbon/human/H in mob_list)
+
+		if(H.iscorpse)
+			continue
+		if(H.z == 2)
+			continue
+
+		var/name
+		var/assignment
+		var/life_status
+		var/list/damage = list()
+		damage["oxygen"] = ""
+		damage["toxin"] = ""
+		damage["burn"] = ""
+		damage["brute"] = ""
+		var/player_area
+		var/see_x = "?"
+		var/see_y = "?"
+		var/see_z = "?"
+
+		// z == 0 means mob is inside object, check is they are wearing a uniform
+		if(istype(H.w_uniform, /obj/item/clothing/under))
+			var/obj/item/clothing/under/U = H.w_uniform
+
+			if (U.has_sensor && U.sensor_mode)
+				var/turf/pos = H.z == 0 || U.sensor_mode == 3 ? get_turf(H) : null
+
+				// Special case: If the mob is inside an object confirm the z-level on turf level.
+				if (H.z == 0 && !pos)
+					continue
+
+				var/obj/item/weapon/card/id/I = H.wear_id ? H.wear_id.GetID() : null
+
+				if (I)
+					name = I.registered_name
+					assignment = I.assignment
+				else
+					name = "Unknown"
+					assignment = ""
+
+				if (U.sensor_mode >= 2)
+					damage["oxygen"] = round(H.getOxyLoss(),1)
+					damage["toxin"] = round(H.getToxLoss(),1)
+					damage["burn"] = round(H.getFireLoss(),1)
+					damage["brute"] = round(H.getBruteLoss(),1)
+
+				//Sensors are at least binary if you are in this block
+				life_status = H.stat //CONSCIOUS, UNCONSCIOUS, DEAD
+				//Crit is only when their damage exceeds their max health
+				if (life_status == 1)
+					//show critical if sensors are 2 or higher, only.
+					if (U.sensor_mode == 1)
+						life_status = 0
+					else if (U.sensor_mode >= 2 && H.health > config.health_threshold_crit)
```

Comment:
Define (DEAD)

Reply:
(SENSORMODE_VITALS), and (CONSCIOUS) below.